### ISSUE TITLE
don't use `UrlFetchTransport` in App Engine Flex environment

### ISSUE THREAD

**Issue by eddavisson** [GoogleCloudPlatform/google-cloud-java#1492]
The decision of whether to use `UrlFetchTransport` instead of `NetHttpTransport` is currently based on the presence of the `com.google.appengine.application.id` system property. This can result in `UrlFetchTranport` being used in the App Engine Flex environment (at least in compat mode).

One possible approach is:

```
if (System.getProperty("com.google.appengine.application.id") != null
    // Rule out App Engine Flex compat mode.
    && System.getenv("GAE_VM") == null
    // Rule out App Engine Flex.
    && System.getenv("GCLOUD_PROJECT" ) == null) {
  // use UrlFetchTransport
}
```

https://github.com/GoogleCloudPlatform/google-cloud-java/blob/e5f90467fe7490759482e6a8bc62b1246318a7f0/google-cloud-core/src/main/java/com/google/cloud/HttpServiceOptions.java#L61

https://github.com/GoogleCloudPlatform/google-cloud-java/blob/0d9d1b3ce548aa5335e23b8c2ec0464df8002d9f/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L318

**Comment by eddavisson** [comment]
Note that a customer seems to be hitting this issue in http://stackoverflow.com/questions/40830727/why-does-datastore-produce-timeout-while-fetching-url.

**Comment by vam-google** [comment]
Should be fixed by https://github.com/GoogleCloudPlatform/google-cloud-java/pull/1893 and will be released with next release (the one **after** 0.12.0). Please reopen if you keep experiencing the issue after the fix.
